### PR TITLE
Python basic module execution unit test on delegation of backend_with_compiler_demo

### PR DESCRIPTION
**Pull request by Amyh11325**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#60468 Python basic module execution unit test on delegation of backend_with_compiler_demo**

Summary:
Added a unit test for the execution of a basic module with a compiler

Test Plan:
Running python test/test_jit.py TestBackendsWithCompiler -v returns a successful test

Reviewers:
myuan

Subscribers:

Tasks:
T91991412

Tags:

Differential Revision: [D29306225](https://our.internmc.facebook.com/intern/diff/D29306225)

### CODE REVIEW

**Comment by Amyh11325** [comment]
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29306225).

File: test/jit/test_backends.py [hunk 1]
```diff
@@ -465,3 +465,126 @@ def test_save_load(self):
...
+    def setUp(self):
+        super().setUp()
+        torch_root = Path(__file__).resolve().parent.parent.parent
+        p = torch_root / 'build' / 'lib' / 'libbackend_with_compiler.so'
+        torch.ops.load_library(str(p))
+        # torch.ops.load_library("//caffe2/test/cpp/jit:test_backend_compiler")
+        # Subclasses are expected to set up three variables in their setUp methods:
+        # module - a regular, Python version of the module being tested
+        # scripted_module - a scripted version of module
+        # lowered_modle - a version of module lowered to a backend
```
Comment (iseeyuan) [inline test/jit/test_backends.py:500]: Remove the commented lines?

File: test/jit/test_backends.py [hunk 2]
```diff
@@ -465,3 +465,126 @@ def test_save_load(self):
     @skipIfRocm
     def test_errors(self):
         self.selective_lowering_test.test_errors()
+
+
+# Unit Tests for backend with compiler
```
Comment (iseeyuan) [inline test/jit/test_backends.py:470]: Add more comments here: 
"The reason to add it as a separate test case, than reusing the existing TestBackends, is these two cases cover different aspects. The actually backend implementation in this test is different. It has a simple demo compiler to test the end-to-end flow. However, this test cannot cover the selective_lowering for now, which is covered in TestBackends."

File: test/jit/test_backends.py [hunk 3]
```diff
@@ -465,3 +465,126 @@ def test_save_load(self):
...
+
+        # The answers returned by Python, JIT and to_backend should all match.
+        self.assertEqual(python_output, backend_output)
+        self.assertEqual(jit_output, backend_output)
+
+    def save_load(self):
+        """
+        Save and load the lowered module.
+        """
+        self.lowered_module = self.getExportImportCopy(self.lowered_module)
```
Comment (iseeyuan) [inline test/jit/test_backends.py:525]: Is this test actually called? Could we add it to TestBackendsWithCompiler()?

**Review by iseeyuan — CHANGES_REQUESTED** [review]
It's a great PR in general! Added some comments inline. Please add save_load tests and update comments. Thanks!

File: test/jit/test_backends.py (hunk 1 above)
Comment (Amyh11325) [reply test/jit/test_backends.py:500]: Including the bottom 4? I just copied those from the existing unit test.

Comment (iseeyuan) [reply test/jit/test_backends.py:500]: Good point. The bottom 4 should be kept.

**Comment by Amyh11325** [comment]
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29306225).

**Comment by Amyh11325** [comment]
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29306225).

File: test/jit/test_backends.py [hunk 4]
```diff
@@ -465,3 +466,121 @@ def test_save_load(self):
     @skipIfRocm
     def test_errors(self):
         self.selective_lowering_test.test_errors()
+
+"""
+Unit Tests for backend with compiler
+This test case and the existing TestBackends are separate because they cover different aspects.
+The actual backend implementation in this test is different.
+It has a simple demo compiler to test the end-to-end flow.
```
Comment (iseeyuan) [inline test/jit/test_backends.py:474]:
```suggestion
It has a simple demo compiler to test the end-to-end flow in mobile.
```

Comment (iseeyuan) [reply test/jit/test_backends.py:474]: Could you also apply clang-format on this PR? You could use `clang-format -i test/jit/test_backends.py` from devserver.

**Review by iseeyuan — APPROVED** [review]
LGTM! Please update the comment and apply clang-format (suggested inline) before landing this PR. Thanks!

**Comment by Amyh11325** [comment]
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29306225).

**Comment by Amyh11325** [comment]
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29306225).

**Pull request closed without merging**